src/scportrait/io/convert.py

```
@@ -27,41 +29,55 @@
+        """
         self.new_project_location = new_project_location
-        self.directory = new_project_location  # This is the directory where the new spatialdata project will be saved and where log output will be saved. This key definition required for the Logable class.
+        self.directory = new_project_location
         self.old_project_location = old_project_location
         self.channel_names = channel_names
         self.cache = cache
         self.debug = debug
         self.overwrite = overwrite
 
-        self.input_image_status = None
-        self.segmentation_status = None
-        self.centers_status = None
+        self.input_image_status: bool | None = None
+        self.segmentation_status: bool | None = None
+        self.centers_status: bool | None = None
 
-    def _check_memory(self, item):
-        """
-        Check the memory usage of the given if it were completely loaded into memory using .compute().
+    def _check_memory(self, item) -> bool:
+        """Check if item can fit in available memory.
+
+        Args:
+            item: Array-like object to check memory requirements for
+
+        Returns:
+            Whether the item can fit in available memory
         """
         array_size = item.nbytes
         available_memory = psutil.virtual_memory().available
-
         return array_size < available_memory
 
-    def _create_temp_dir(self):
-        """
-        Create a temporary directory in the cache directory specified in the config for saving all intermediate results.
-        """
+    def _create_temp_dir(self) -> None:
+        """Create a temporary directory for intermediate results.
 
+        Creates directory in cache location specified in config or current working directory.
+        """
         if self.cache is not None:
-            path = os.path.join(self.config["cache"], f"{self.__class__.__name__}_")
+            path = os.path.join(self.directory["cache"], f"{self.__class__.__name__}_")  # type: ignore
```

self.directory should raise an error I think because it doesn't exist. Was there a specific reason you changed it?

---

Yes, `self.config` does not exist. What should it be?

---

should be self.cache. I fixed it in another commit. Not sure what happened there. Thanks for catching it.